app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayInfoBannerBinding.kt

```diff
@@ -0,0 +1,123 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabstray
+
+import android.content.Context
+import android.view.View.VISIBLE
+import android.view.ViewGroup
+import androidx.annotation.VisibleForTesting
+import kotlin.math.max
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.map
+import mozilla.components.browser.state.selector.normalTabs
+import mozilla.components.browser.state.selector.privateTabs
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+import org.mozilla.fenix.R
+import org.mozilla.fenix.browser.infobanner.InfoBanner
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.components.metrics.MetricController
+import org.mozilla.fenix.utils.Settings
+
+class TabsTrayInfoBannerBinding(
+    private val context: Context,
+    private val store: BrowserStore,
+    private val infoBannerView: ViewGroup,
+    private val settings: Settings,
+    private val navigationInteractor: NavigationInteractor,
+    private val metrics: MetricController?
+) : LifecycleAwareFeature {
+    private var scope: CoroutineScope? = null
+
+    @VisibleForTesting
+    internal var testBanner: InfoBanner? = null
```

Comment:
I'm not super keen about adding code just for testing. We could either change our implementation to make it more testable or we change what we're testing for.

We could try to look at the child count of the parent `infoBannerView` group and make assertions on the child to see which banner is up. We can assert the `visibility` to see if any banner is shown.

Reply:
Sure, I'll update the test.

Reply:
Let's also make it plural: _tests_. 😄

Reply:
That was assumed 😆